audio/audiodata.c

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2017 Kyle Robbertze <krobbertze@gmail.com>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+/** \file audiodata.c
+ * Contains definitions for USB audio communication between the FX2 chip and
+ * the host
+ */
+
+#include <stdio.h>
+#include <fx2regs.h>
+#include <delay.h>
+#include <eputils.h>
+
+#include "audiodata.h"
+
+#define SYNCDELAY SYNCDELAY4
+
+/**
+ * Returns the configuration. We only support cfg 1
```

Comment:
Why do you only support cfg 1?

Reply:
Following the video firmware

Reply:
Why does the video firmware do that? :-P

Reply:
Looking at the descriptors, we only have one configuration - USB audio device for audio firmware and USB video device for the video firmware. So we can only work with that config. Reference: http://www.beyondlogic.org/usbnutshell/usb5.shtml#ConfigurationDescriptors

I'll update the comment

Reply:
Fixed